jme3-glsl-highlighter/src/com/jme3/gde/glsl/highlighter/lexer/GlslKeywordLibrary.java

```diff
@@ -0,0 +1,450 @@
+package com.jme3.gde.glsl.highlighter.lexer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Brace, yourselves, this file contains every word that means something in GLSL.
+ * Expect about 400 lines of code that just adds strings.
+ * @author grizeldi
+ */
+class GlslKeywordLibrary {
+    public enum KeywordType{
+        KEYWORD, BUILTIN_FUNCTION, BUILTIN_VARIABLE, BASIC_TYPE, UNFINISHED;
+    }
+    private static final List<String> keywords = new ArrayList<>(),
+            builtinFunctions = new ArrayList<>(),
+            builtinVariables = new ArrayList<>(),
+            basicTypes = new ArrayList<>();
+    
+    static {
```

Comment:
I'm not sure if the Array notation in combination with Arrays.asList() would be easier/more compact and maybe even faster. But for this PR, I don't care about that. Just something to keep in mind if you have to work on this file for code completion or something.
But yeah, I'm not picky on that one.